lib/blpapi-wrapper.ts

```diff
@@ -336,9 +353,18 @@ export class Session extends events.EventEmitter {
                 // XXX: yes, this is a side-effect of map, but it is needed for performance reasons
                 //      until ES6 Map is available
                 this.subscriptions[cid] = s;
-                this.listen('MarketDataEvents', cid, (m: any): void => {
-                    s.emit('data', m.data, s);
-                });
+
+                // Lookup by service the events we need to listen to.
+                if (!(s.serviceUri in SERVICE_TO_SUBSCRIPTION_EVENTS_MAP)) {
+                    throw new Error('Unknown service: ' + s.serviceUri);
+                }
+                var events: string[] = SERVICE_TO_SUBSCRIPTION_EVENTS_MAP[s.serviceUri];
+                for (var i: number = 0 ; i < events.length ; i++) {
```

Comment:
No reason on my part other than it seemed overkill to set a callback for what needs to be done in the body of the loop or is there a forEach that doesn't take a callback?